crates/cairo-serde-derive/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use proc_macro::{self, TokenStream};
+use quote::quote;
+use syn::{parse_macro_input, Data, DeriveInput, Ident, Type};
+
+#[proc_macro_derive(CairoSerde)]
+pub fn derive(input: TokenStream) -> TokenStream {
+    let DeriveInput { ident, data, .. } = parse_macro_input!(input);
+
+    let (idents, types): (Vec<Ident>, Vec<Type>) = match data {
+        Data::Struct(data) => data
+            .fields
+            .iter()
+            .cloned()
+            .map(|field| (field.ident.clone().unwrap(), field.ty))
+            .unzip(),
+        _ => panic!("CairoSerde can only be derived for structs"),
+    };
+
+    let cairo_serialized_size = quote! {
+        fn cairo_serialized_size(rust: &Self::RustType) -> usize {
+            0
+            #(
+                + <#types as ::cainome_cairo_serde::CairoSerde>::cairo_serialized_size(&rust.#idents)
+            )*
+        }
+    };
+
+    let cairo_serialize = quote! {
+        fn cairo_serialize(rust: &Self::RustType) -> Vec<Felt> {
```

Comment:
```suggestion
        fn cairo_serialize(rust: &Self::RustType) -> Vec<starknet::core::types::Felt> {
```
Use fully qualified path as possible to avoid downstream usage to include all the types manually.

Reply:
fixed